core/src/main/java/org/infinispan/factories/threads/DefaultNonBlockingThreadFactory.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.factories.threads;
+
+import org.infinispan.commons.executors.NonBlockingThread;
+import org.infinispan.commons.executors.NonBlockingThreadFactory;
+
+public class DefaultNonBlockingThreadFactory extends DefaultThreadFactory implements NonBlockingThreadFactory {
```

Comment:
I think it should be just `NonBlockingThreadFactory`, or we could even replace it with a boolean in `DefaultThreadFactory`.

Reply:
I can rename it.

Reply:
Actually I can't rename it, since the interface is that name :)

Reply:
I forgot about the interface. I'm having doubts about the change now, because we're breaking backwards compatibility, but I guess `ThreadFactory` isn't customized that often so it will mostly affect @pferraro